game/src/forge/mod.rs

```diff
@@ -278,7 +278,10 @@ unsafe extern "system" fn wnd_proc(window: HWND, msg: u32, wparam: usize, lparam
                     });
                     if let Some(ref window) = forge.window {
                         // Move the cursor to the middle of the window
-                        SetCursorPos(window.width / 2, window.height / 2);
+                        SetCursorPos(
+                            window.client_x + window.width / 2,
```

Comment:
Not sure this does what it should be doing. In any case, it doesn't center the cursor for me. It only centers it *after* I exit the game, and only if I didn't move it until then.
Why not just hard code it to 320/240 like it was before? - https://github.com/ShieldBattery/ShieldBattery/blob/master/game/forge/forge.cpp#L594

Reply:
Oh, right, you are correct.

It only works currently if the mouse sensitivity setting is set to leftmost value, which I had set.
Note that the C++ code calls the hook, which does the conversion from BW coords to screen coords, while this called the real function directly.

The reason don't want to call the hook is that Rust doesn't allow getting mutable access to a single variable simultaneously from two separate points, e.g.
```rust
fn parent() {
  // Access the global forge variable
  with_forge(|forge| {
    // do things...
    something(320, 240);
  });
}

fn something(x: i32, y: i32) {
  // Accesses the global forge variable, but parent function is already doing that :(
  with_forge(|forge| {
    // ...
  });
}
```
is not going to work.

Just calling that child function outside `with_forge` would obviously work, but trying to keep track of which helper functions can be called and which can not is likely going to end up being a nightmare. So I've kept any `with_forge` call directly at the hook functions themself, not calling any of those hooks myself, and any helper function needing forge state takes it as an `&mut Forge` parameter.

So I figured it's not worth it to create a helper function to share the logic between SetCursorPos hook and this one time we explicitly center screen at 320, 240 - the center of window, simple enough :) . ..But yeah, I didn't think it all the way through, there's enough logic needed that having a function `forge.set_cursor_to_bw_coords(320, 240)`, and having the hook call that as well seems to be the best way to do it.